### PR TITLE
Adding PyTorch Frontend pdist #11753

### PR DESCRIPTION
**Pull request by mrdedede**
Created Frontend PyTorch pdist distance algorithm
Related to #11753 and #25

### CODE REVIEW

**Comment by zaeemansari70** [comment]
Hi @mrdedede  ,
Do request a review from me by just commenting on the PR once you are done with the implementation slightly_smiling_face

**Comment by mrdedede** [comment]
Hi @zaeemansari70 
Thanks!
I'll look forward to check what the failing tests may mean.

Meanwhile, would you mind helping me understand some of the error messages? Most of them are just one line long and don't offer much advice on what does need changes.

File: ivy/functional/frontends/torch/nn/functional/distance_functions.py [hunk 1]
```diff
@@ -40,3 +40,9 @@ def pairwise_distance(x1, x2, *, p=2.0, eps=1e-06, keepdim=False):
         output_dim = x2_dim
 
     return ivy.vector_norm(x1 - x2 + eps, ord=p, axis=output_dim - 1, keepdims=keepdim)
+
+@with_unsupported_dtypes({"1.11.0 and below": ("float16", "bfloat16")}, "torch")
+@to_ivy_arrays_and_back
+def pdist(x1, p=2.0):
+    x1 = torch_frontend.promote_types_of_torch_inputs(x1)
```
Comment (zaeemansari70) [inline ivy/functional/frontends/torch/nn/functional/distance_functions.py:47]: You are using `promote_types_of_torch_inputs()` which takes two arguments and you passing one. :slightly_smiling_face: 
https://github.com/unifyai/ivy/blob/master/ivy/functional/frontends/torch/__init__.py#L241

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
I've mentioned the issue n your code below.
Thanks! :slightly_smiling_face:

**Comment by mrdedede** [comment]
Thanks, @zaeemansari70 
I'll look into it asap! :)

I have had some issues trying to run my local tests, however, I can't seem to get it right when I run it locally.
Do you mind if I ping you on Discord in order to analyze why this may be happening? Testing locally and getting my results there would make a huge difference!!

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**